### ISSUE TITLE
Is their a reason @mui/x-date-pickers is in my bundle when I don't have it enabled or imported? 

### ISSUE THREAD

**Issue by MylesWardell** [KevinVandy/material-react-table#1328]
### material-react-table version

3.0.1

### react & react-dom versions

19.0.0

### Describe the bug and the steps to reproduce it

I am using vite with the rollup-plugin-visualizer. Just curious if this is intended or if I have misconfigured the table? 

### Minimal, Reproducible Example - (Optional, but Recommended)

1. Install and use material react table with a bundle analyzer
2. See that @mui/x-date-pickers and its full bundle is included 

### Screenshots or Videos (Optional)

![image](https://github.com/user-attachments/assets/15ab2e66-9f4e-42e1-a927-5341871855cf)

### Do you intend to try to help solve this bug with your own PR?

None

### Terms

- [X] I understand that if my bug cannot be reliably reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by KevinVandy** [comment]
MRT imports it

**Comment by MylesWardell** [comment]
@KevinVandy maybe a feature request then? Why is this in the bundle? Can it be injected instead?

**Comment by KevinVandy** [comment]
Maybe in the future

**Comment by MylesWardell** [comment]
@KevinVandy I tried to pull the repo and find the leak in your imports (but its not particularly friendly to the PowerShell users). It defiantly feels like more is getting imported and somewhere their is a `mui/x-date-pickers` without the tail IE (@mui/x-date-pickers/AdapterDayjs). @mui/material looks like it is not importing everything so maybe some kind of rollup config? 

I could not find the issue but my one guess is this grouping in the export area. Unfamiliar with the rollup config that you have so I would have to poke around more to find out

![image](https://github.com/user-attachments/assets/ea42642d-266d-48fc-9466-951653eacfdb)

TLDR: Regardless pretty sure that even with using `@mui/x-date-pickers` your three imports should not import everything else

![image](https://github.com/user-attachments/assets/fe2c0fb9-6c56-4d5e-b4f1-caa04752f18f)

**Comment by MylesWardell** [comment]
Experimenting with config options on my own rollup but their potentially is a import leak here

node_modules\material-react-table\dist\index.d.ts
![image](https://github.com/user-attachments/assets/3d010cde-3bfc-4e0f-92d7-2c418bad32b1)

I cant see any other reason for the whole package to be imported

**Comment by KevinVandy** [comment]
Your prod bundle should only be importing the actual components used by MRT, not the entire library. Are you saying that's what's happening?

**Comment by MylesWardell** [comment]
Yes that's what's happening. Sounds like it might be an issue that only I am facing? I'll pull up a separate repo end experiment when I get back from holiday

**Comment by rob2d** [comment]
> Experimenting with config options on my own rollup but their potentially is a import leak here
> 
> node_modules\material-react-table\dist\index.d.ts ![image](https://github.com/user-attachments/assets/3d010cde-3bfc-4e0f-92d7-2c418bad32b1)
> 
> I cant see any other reason for the whole package to be imported

Can confirm this is an issue - x-date-pickers alone is 471kb.
![Image](https://github.com/user-attachments/assets/d3041762-6202-498f-99e4-2bf3062d3798)

In a minimalist example alone of only rendering MRT + grid and no other MUI components and < 10 rows of data + 5 columns (following the most basic example), I get a bundle size of 521kb (146kb gzip).

![Image](https://github.com/user-attachments/assets/e0de9278-fef7-4585-a18e-3830202a6dea)

Mui-modern preference does not seem to help with tree shaking so the only thing that can be done to help at all is lazy-loading.

**Comment by MylesWardell** [comment]
The mui-date-picker imports unpacked are ~twice the size of material-react-table but the library has a couple of solutions
1. Use whatever system dayjs/chartjs plugins use to add additional functionality (should be some kind of es module magic)
1. lazy import table functions, which has the trade off of a wait time on those features being used for first time. 
2. Create a separate build that just does not include the date-picker features that can be pointed to with `material-react-table/mini`

Its not a large impact as you can lazy load the tables in but it would be good for the long term health of the package 👍

**Comment by singpolyma** [comment]
If I comment out these lines from `node_modules/material-react-table/dist/index.esm.js`

```
//import { DatePicker } from '@mui/x-date-pickers/DatePicker';                                                                                                               
//import { DateTimePicker } from '@mui/x-date-pickers/DateTimePicker';                                                                                                       
//import { TimePicker } from '@mui/x-date-pickers/TimePicker'; 
```

then my bundle size goes down by 219kb. I assume this breaks something if I was using the date picker features but I don't think I am.
